processor/processor_test.go

```diff
@@ -0,0 +1,71 @@
+package processor
+
+import (
+	"os"
+	"testing"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/util/test"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInit(t *testing.T) {
+	genesis := test.MakeGenesis()
+	genesisBlock := test.MakeGenesisBlock()
+	dir := "/tmp/ledger"
+	err := os.Mkdir(dir, 0755)
+	defer os.RemoveAll(dir)
+	assert.Nil(t, err)
+
+	processor := processorImpl{}
+	err = processor.Init(dir, &genesis, &genesisBlock)
+	assert.Nil(t, err)
+	assert.NotNil(t, processor.ledger)
+}
+
+func TestProcess(t *testing.T) {
```

Comment:
Not that this test is especially complicated, but Tolik's suggestion should make it even simpler.

The separate Init tests can be done against against `MakeBlockProcessor`.
Process tests can focus on the logic using an in-memory ledger.